test/integration/Bijectors/runtests.jl

```diff
@@ -0,0 +1,222 @@
+module BijectorsIntegrationTests
+
+using Bijectors: Bijectors
+using Enzyme: Enzyme
+using FiniteDifferences: FiniteDifferences
+using LinearAlgebra: LinearAlgebra
+using Random: randn
+using StableRNGs: StableRNG
+using Test: @test, @test_broken, @testset
+
+rng = StableRNG(23)
+
+"""
+Enum type for choosing Enzyme autodiff modes.
+"""
+@enum ModeSelector Neither Forward Reverse Both
+
+"""
+Type for specifying a test case for `Enzyme.gradient`.
+
+The test will check the accuracy of the gradient of `func` at `value` against `finitediff`,
+with both forward and reverse mode autodiff. `name` is for diagnostic printing.
+`runtime_activity`, `broken`, `skip` are for specifying whether to use
+`Enzyme.set_runtime_activity` or not, whether the test is broken, and whether the test is so
+broken we can't even run `@test_broken` on it (because it crashes Julia). All of them take
+values `Neither`, `Forward`, `Reverse` or `Both`, to specify which mode to apply the setting
+to. `splat` is for specifying whether to call the function as `func(value)` or as
+`func(value...)`.
+
+Default values are `name=nothing`, `runtime_activity=Neither`, `broken=Neither`,
+`skip=Neither`, and `splat=false`.
+"""
+struct TestCase
+    func::Function
+    value
+    name::Union{String, Nothing}
+    runtime_activity::ModeSelector
+    broken::ModeSelector
+    skip::ModeSelector
+    splat::Bool
+end
+
+# Default values for most arguments.
+function TestCase(
+    f, value;
+    name=nothing, runtime_activity=Neither, broken=Neither, skip=Neither, splat=false
+)
+    return TestCase(f, value, name, runtime_activity, broken, skip, splat)
+end
+
+"""
+Test Enzyme.gradient, both Forward and Reverse mode, against FiniteDifferences.grad.
+"""
+function test_grad(case::TestCase; rtol=1e-6, atol=1e-6)
+    @nospecialize
+    f = case.func
+    # We'll call the function as f(x...), so wrap in a singleton tuple if need be.
+    x = case.splat ? case.value : (case.value,)
+    finitediff = collect(
+        FiniteDifferences.grad(FiniteDifferences.central_fdm(4, 1), f, x...)[1]
+    )
+
+    f_mode = if (case.runtime_activity === Both || case.runtime_activity === Forward)
+        Enzyme.set_runtime_activity(Enzyme.Forward)
+    else
+        Enzyme.Forward
+    end
+    r_mode = if (case.runtime_activity === Both || case.runtime_activity === Reverse)
+        Enzyme.set_runtime_activity(Enzyme.Reverse)
+    else
+        Enzyme.Reverse
+    end
+
+    if !(case.skip === Forward) && !(case.skip === Both)
+        if case.broken === Both || case.broken === Forward
+            @test_broken(
+                collect(Enzyme.gradient(f_mode, Enzyme.Const(f), x...)[1]) ≈ finitediff,
```

Comment:
why are these collect's needed?

Reply:
They are not. Removed. Thanks for spotting.